core/test/Test/Pos/Core/Bi.hs

```diff
@@ -0,0 +1,1739 @@
+
+roundTripTxSizeLinearBi :: Property
+roundTripTxSizeLinearBi = eachOf 1000 genTxSizeLinear roundTripsBiBuildable
+
+--------------------------------------------------------------------------------
+-- TxWitness
+--------------------------------------------------------------------------------
+
+golden_TxWitness :: Property
+golden_TxWitness = goldenTestBi exampleTxWitness "test/golden/TxWitness"
+
+roundTripTxWitness :: Property
+roundTripTxWitness = eachOf 20 (feedPM genTxWitness) roundTripsBiShow
+
+--------------------------------------------------------------------------------
+-- UpAttributes
+--------------------------------------------------------------------------------
+
+golden_UpAttributes :: Property
+golden_UpAttributes = goldenTestBi exampleUpAttributes "test/golden/UpAttributes"
+
+roundTripUpAttributes :: Property
+roundTripUpAttributes = eachOf 20 genUpAttributes roundTripsBiBuildable
+
+--------------------------------------------------------------------------------
+-- UpdateData
+--------------------------------------------------------------------------------
+
+golden_UpdateData :: Property
+golden_UpdateData = goldenTestBi exampleUpdateData "test/golden/UpdateData"
+
+roundTripUpdateData :: Property
+roundTripUpdateData = eachOf 20 genUpdateData roundTripsBiBuildable
+
+--------------------------------------------------------------------------------
+-- UpdatePayload
+--------------------------------------------------------------------------------
+
+golden_UpdatePayload :: Property
+golden_UpdatePayload = goldenTestBi exampleUpdatePayload "test/golden/UpdatePayload"
+
+roundTripUpdatePayload :: Property
+roundTripUpdatePayload = eachOf 20 (feedPM genUpdatePayload) roundTripsBiBuildable
+
+--------------------------------------------------------------------------------
+-- UpdateProof
+--------------------------------------------------------------------------------
+
+golden_UpdateProof :: Property
+golden_UpdateProof = goldenTestBi exampleUpdateProof "test/golden/UpdateProof"
+
+roundTripUpdateProof :: Property
+roundTripUpdateProof = eachOf 20 (feedPM genUpdateProof) roundTripsBiBuildable
+
+--------------------------------------------------------------------------------
+-- UpdateProposal
+--------------------------------------------------------------------------------
+
+golden_UpdateProposal :: Property
+golden_UpdateProposal = goldenTestBi exampleUpdateProposal "test/golden/UpdateProposal"
+
+roundTripUpdateProposal :: Property
+roundTripUpdateProposal = eachOf 20 (feedPM genUpdateProposal) roundTripsBiBuildable
+
+--------------------------------------------------------------------------------
+-- UpdateProposals
+--------------------------------------------------------------------------------
+
+golden_UpdateProposals :: Property
+golden_UpdateProposals = goldenTestBi ups "test/golden/UpdateProposals"
+  where
+    -- Need to revisit this.
+    ups = HM.fromList [(exampleUpId, exampleUpdateProposal)]
```

Comment:
This could use more list elements. Something like [`examplePublicKeys`](https://github.com/input-output-hk/cardano-sl/pull/3120/files#diff-6e9537b3a5b17a440b3dfd4f7c1b9715R1503), to give us lists of datatypes for golden tests, might be good.

Reply:
Yeah exactly. I noted to revisit this because I wanted to add some "static" data to use in this list. This is something I can get done.